### ISSUE TITLE
Global color custom name for new colors

### ISSUE THREAD

**Issue by kskonovalov** [elementor/elementor#14560]
- [x] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [x] The feature is still missing in the latest stable version of Elementor ( Elementor Pro. )

**What problem is your feature request going to solve? Please describe.**
There's an amazing option to define the global colors, and often I need to use them in my css files like **color: var( --e-global-color-accent );**
The problem is, only the first 4 (pre-defined) types of colors has the human friendly names like **--e-global-color-accent**, but for the newly created colors names are generated like **--e-global-color-74c5eff**

**Describe the solution you'd like**
Would be awesome if it's possible to name the variable the same way as there's the name for the color

Screenshot of what I mean https://ibb.co/VQtYXML

**Comment by stasonua0** [comment]
A much needed revision, +1

**Comment by black-eye** [comment]
+1

**Comment by euene** [comment]
+1

**Comment by MichaelDarko** [comment]
Yes! The random values are a problem. We now have global colors, so we will naturally use them in our custom CSS all the time. When the names are not "human friendly", it's impossible to just type the variable value because you can't remember the random numbers and letters. This results into constant use of the inspect tool to copy-paste the values. 

It also makes it hard to read the custom CSS because you don't know which color is which. If we could set the var name ourselves, that would make life so much easier since you have to write a ton of custom CSS to overcome the lack of Elementor's styling options.

(And BTW, the custom CSS editor doesn't recognize variable values and is showing warning when you use them. Bit annoying and should be updated.)

**Comment by Sitecentrale** [comment]
Indeed much needed (and duplicate of #12571 ?)

**Comment by kskonovalov** [comment]
> Indeed much needed (and duplicate of #12571 ?)

oops you right, I tried to search but missed it when I asked

**Comment by gerardreches** [comment]
Here is **another reason to support this** request:

When exporting and importing templates to another site, if there were some non-default global colors assigned to widgets, they won't be found even if the new site has them defined with the same names.

Example:

In every site I work on, I have the following **global colors**:

- Primary
- Secondary
- Text
- Accent
- Black
- Dark gray
- Gray
- Light gray
- White
- Transparent

When I use, for example, the dark gray global color in a widget included in a template, and then I export this template and import it into another of my sites, the global color binding will be lost since due to the random generated name it won't match the new site's global colors. If the name wasn't randomly generated, it would match and the binding would not break.

This is a big deal for exportable templates.

**Comment by iMiek** [comment]
+1

**Comment by forrestkirby** [comment]
I would love to see this feature added to Elementor, too. It would simplify the use of global colora in custom CSS.

**Comment by paauserrat** [comment]
This is needed

**Comment by rami-elementor** [comment]
@kskonovalov 

What will happened if someone will change the color variable name?

**Comment by kskonovalov** [comment]
@rami-elementor 
I think, it's good to leave the color name like it was generated originally. Then all pages which are already using this color wouldn't loose it.

So if generate the color name with it's title, the only difference with the current option, will be user-friendly ' --e-global-color-second-accent' instead of '--e-global-color-74c5eff'.

And if user needs the new name, he can just create the new color, that's all

**Comment by satindersinghsidhu** [comment]
I'm already giving custom names to global color variables by using below method:-

**Suppose you have below variables from elementor global color:-**
```
.elementor-kit-5 {
--e-global-color-primary: #000000;
--e-global-color-secondary: #A80D52;
--e-global-color-text: #000000;
--e-global-color-accent: #A80D52;
--e-global-color-7deaba8: #FCFCF7;
--e-global-color-724613d: #FFFFFF;
--e-global-color-9bc59b2: #52BCBA;
--e-global-color-27f5279: #515151;
--e-global-color-39be410: #DFDCDE;
--e-global-color-449d36b: #097512;
}
```

**Now define your own custom variables inside custom css like below:-**
```
.elementor-kit-5 {
--e-my-custom-name1: var(--e-global-color-primary);
--e-my-custom-name2: var(--e-global-color-secondary);
--e-my-custom-name3: var(--e-global-color-text);
--e-my-custom-name4: var(--e-global-color-accent);
--e-my-custom-name5: var(--e-global-color-7deaba8);
--e-my-custom-name6: var(--e-global-color-724613d);
--e-my-custom-name7: var(--e-global-color-9bc59b2);
--e-my-custom-name8: var(--e-global-color-27f5279);
--e-my-custom-name9: var(--e-global-color-39be410);
--e-my-custom-name10: var(--e-global-color-449d36b);
}
```

**Comment by kskonovalov** [comment]
@satindersinghsidhu you don't need to have .elementor-kit-5 class for it. You can use it actually, however it's not necessary ([https://developer.mozilla.org/en-US/docs/Web/CSS/--*](https://developer.mozilla.org/en-US/docs/Web/CSS/--*)).
The common recipe is
:root {
  --main-bg-color: brown;
}

But the idea was to manage the colors both inside the elementor editor and in the css files

**Comment by about-hector** [comment]
@satindersinghsidhu hey man, may I ask you where do you do that?

do you make a child theme and overwrite the styles.css?

use site settings > global css? 

appearence > css ?

which is the best way, in your opinion, to declare global variables on elementor and be able to reuse them with no plugins and useless stuff?

**Comment by phkoon** [comment]
This alone would solve a big issue in the coexistence of v3 and v4
